owncloudApp/src/main/java/com/owncloud/android/ui/dialog/BaseDialogFragment.kt

```diff
@@ -0,0 +1,35 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author David Crespo Ríos
+ * Copyright (C) 2022 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.ui.dialog
+
+import android.app.Dialog
+import android.view.WindowManager
+import androidx.fragment.app.DialogFragment
+import com.owncloud.android.BuildConfig
+import com.owncloud.android.R
+
+open class BaseDialogFragment : DialogFragment() {
+
+    fun avoidScreenshotsIfNeeded(dialog: Dialog) {
+        if (!BuildConfig.DEBUG && context?.resources?.getBoolean(R.bool.allow_screenshots) == false) {
+            dialog.window?.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
+        }
+    }
+}
```

Comment:
`Base` classes are sometimes a good idea, not always. But in this case, is absolutely not needed. You could create a random class and that function would work since it does not perform anything related to itself nor overrides anything. It receives the dialog from the parameter and applies that flag.

You could instead create a Kotlin extension:
```
fun Dialog.avoidScreenshotsIfNeeded() {
    if (!BuildConfig.DEBUG && context.resources?.getBoolean(R.bool.allow_screenshots) == false) {
        window?.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
    }
}
```

And call it like this `dialog.avoidScreenshotsIfNeeded()` with no need of base classes.

What do you think?